pyxrf/model/fileio.py

```diff
@@ -585,35 +660,40 @@ def output_data_to_tiff(fit_output,
         tiff or txt
     name_append: str, optional
         more information saved to output file name
-    norm_name : str, optional
+    scaler_name : str, optional
         if given, normalization will be performed.
     use_average : Bool, optional
-        when normalization, multiply mean value of denomenator,
-        i.e., norm_data = data1/data2 * np.mean(data2)
+        when normalization, multiply by the mean value of scaler,
+        i.e., norm_data = data/scaler * np.mean(scaler)
     """
     # save data
     if os.path.exists(output_folder) is False:
-        logger.warning("Output_folder {} is created".format(output_folder))
         os.mkdir(output_folder)
 
-    if norm_name is not None:
-        ic_v = fit_output[str(norm_name)]
-        norm_sign = 'norm'
-        for k, v in six.iteritems(fit_output):
-            if 'pos' in k or 'r2' in k:
-                continue
-            ave = 1.0
-            if use_average is True:
-                ave = np.mean(ic_v)
-            v = v/ic_v * ave
-            _fname = "_".join([k, name_append, norm_sign])
-            if file_format == 'tiff':
-                fname = os.path.join(output_folder, _fname + '.tiff')
-                sio.imsave(fname, v.astype(np.float32))
-            elif file_format == 'txt':
-                fname = os.path.join(output_folder, _fname + '.txt')
-                np.savetxt(fname, v.astype(np.float32))
+    # Normalize data if scaler is provided
+    if scaler_name is not None:
+        if scaler_name in fit_output:
+            ic_v = fit_output[scaler_name]
+            norm_sign = 'norm'
+            for k, v in six.iteritems(fit_output):
+                if 'pos' in k or 'r2' in k:
+                    continue
+                # Normalization of data
+                v = normalize_data_by_scaler(v, ic_v)
+                if use_average is True:
+                    v *= np.mean(ic_v)
+                _fname = "_".join([k, name_append, norm_sign])
+                if file_format == 'tiff':
+                    fname = os.path.join(output_folder, _fname + '.tiff')
```

Comment:
I am not sure I understand how the code can be optimized without making it obscure. In this function we create 4 different file names. Adding another ``if...else`` statement to set the variable holding file extension does not add clarity, just increases the number of code lines. Also this is a little bit rearranged code from old PyXRF and written in style of old PyXRF.

Reply:
It's to satisfy the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle. If you or someone else decides to change the `output_folder` name to something else, it will have to be done in 2 places. If there are new formats added, you will have to repeat it N times. I think the following snippet is better future-proof:
```py
file_format = file_format.lower()
_fname = os.path.join(output_folder, f'{_fname}.{file_format}')
if file_format == 'tiff':
    sio.imsave(fname, v.astype(np.float32))
elif file_format == 'txt':
    np.savetxt(fname, v.astype(np.float32))
elif file_format == '<my new format>':
    <format's saving function>(fname, ...)
elif file_format == '<another new format>':
    <another format's saving function>(fname, ...)
else:
    raise RuntimeError(f'The format "{file_format}" is not supported yet.')
```

Reply:
I think a changed the function in a way that makes the most sense, while unlinking extension names from 'format IDs'. I don't think that arbitrary chosen IDs should be used as extension names, since they match only by chance.